src/BeaconConfig.jsx

```diff
@@ -28,4 +57,33 @@ export default class BeaconConfig extends Component {
       </div>
     );
   }
+
+  storeHash(hash) {
+    const transaction = this.state.database.transaction(['beacons'], 'readwrite');
+    transaction.objectStore('beacons').add(true, hash);
+
+    this.setState({ dbCache: { ...this.state.dbCache, [hash]: hash }});
+  }
+
+  loadHash(hash, callback) {
```

Comment:
Callback is invoked with `true` if the hash is found (i.e. the user has already clicked on the beacon) and `false` otherwise.